pkg/catalog/routes.go

```diff
@@ -381,19 +364,40 @@ func getDefaultWeightedClusterForService(meshService service.MeshService) servic
 }
 
 // listTrafficTargetPermutations creates a list of TrafficTargets for each source and destination pair.
-func listTrafficTargetPermutations(name string, srcServiceList []service.MeshService, destServiceList []service.MeshService) []trafficpolicy.TrafficTarget {
+func (mc *MeshCatalog) listTrafficTargetPermutations(trafficTarget target.TrafficTarget, src target.IdentityBindingSubject, dest target.IdentityBindingSubject) ([]trafficpolicy.TrafficTarget, error) {
+	sourceServiceAccount := service.K8sServiceAccount{
+		Namespace: src.Namespace,
+		Name:      src.Name,
+	}
+
+	srcServiceList, srcErr := mc.GetServicesForServiceAccount(sourceServiceAccount)
+	if srcErr != nil {
+		log.Error().Msgf("TrafficTarget %s/%s could not get source services for service account %s", trafficTarget.Namespace, trafficTarget.Name, sourceServiceAccount.String())
+		return nil, srcErr
+	}
+
+	dstNamespacedServiceAcc := service.K8sServiceAccount{
+		Namespace: dest.Namespace,
+		Name:      dest.Name,
+	}
+	destServiceList, destErr := mc.GetServicesForServiceAccount(dstNamespacedServiceAcc)
+	if destErr != nil {
+		log.Error().Msgf("TrafficTarget %s/%s could not get destination services for service account %s", trafficTarget.Namespace, trafficTarget.Name, dstNamespacedServiceAcc.String())
+		return nil, destErr
+	}
+
 	trafficPolicies := make([]trafficpolicy.TrafficTarget, 0, len(srcServiceList)*len(destServiceList))
 
 	for _, destService := range destServiceList {
 		for _, srcService := range srcServiceList {
 			trafficTarget := trafficpolicy.TrafficTarget{
```

Comment:
local scope keeps it safe, but just to avoid confusion between the param and the local var maybe:
```suggestion
			tempTrafficTarget := trafficpolicy.TrafficTarget{
				Name:        utils.GetTrafficTargetName(trafficTarget.Name, srcService, destService),
				Destination: destService,
				Source:      srcService,
			}
			trafficPolicies = append(trafficPolicies, tempTrafficTarget)
```

Reply:
Agreed. I will update this in a different PR I'm working on.